### PR TITLE
adds k8s package for k8s setup

### PR DESCRIPTION
**Pull request by xh3b4sd**

### CODE REVIEW

File: k8s/setup.go [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+)
+
+type SetupConfig struct {
+	K8sClient kubernetes.Interface
+	Logger    micrologger.Logger
+}
+
+type Setup struct {
```
Comment (kopiczko) [inline k8s/setup.go:21]: It's more of a `Client` IMO.

File: k8s/setup.go [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+		return nil
+	}
+	b := backoff.NewExponential(backoff.ShortMaxWait, backoff.ShortMaxInterval)
+
+	err := backoff.Retry(o, b)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	s.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("ensured Kubernetes namespace %#q", namespace))
```
Comment (kopiczko) [inline k8s/setup.go]: When `Kubernetes` is capitalized then I think  `Namespace` should also be.

**Review by kopiczko — APPROVED** [review]

File: k8s/setup.go (hunk 1 above)
Comment (xh3b4sd) [reply k8s/setup.go:21]: `k8s.Setup` reads better than `k8s.Client`. The latter is misleading IMO.

File: k8s/setup.go (hunk 2 above)
Comment (xh3b4sd) [reply k8s/setup.go]: Done.

File: k8s/setup.go (hunk 1 above)
Comment (kopiczko) [reply k8s/setup.go:21]: OK. Go ahead.

**Pull request merged**